pkg/kallsyms/kallsyms_test.go

```diff
@@ -224,3 +226,258 @@ func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
 		require.Equal(b, true, exists)
 	}
 }
+
+func TestNewKAllSymsFromReaderErrors(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expectedErr string
+	}{
+		{
+			name:        "invalid_format_less_fields",
+			input:       "0000000000000000 A",
+			expectedErr: "line \"0000000000000000 A\" has less than 3 fields",
+		},
+		{
+			name:        "invalid_address",
+			input:       "INVALID_ADDR A fixed_percpu_data",
+			expectedErr: "strconv.ParseUint",
+		},
+		{
+			name:        "malformed_line",
+			input:       "0000000000000000",
+			expectedErr: "line \"0000000000000000\" has less than 3 fields",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			_, err := NewKAllSymsFromReader(reader)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tt.expectedErr)
+		})
+	}
+}
+
+func TestLookupByInstructionPointerEdgeCases(t *testing.T) {
+	kAllSymsStr := strings.Join([]string{
+		"0000000000000000 A first_symbol",
+		"0000000000001000 A second_symbol",
+		"0000000000002000 A third_symbol",
+		"0000000000003000 A fourth_symbol",
+	}, "\n")
+
+	kAllSymsReader := strings.NewReader(kAllSymsStr)
+	kAllSyms, err := NewKAllSymsFromReader(kAllSymsReader)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		ip       uint64
+		expected string
+	}{
+		{
+			name:     "exact_match_start",
+			ip:       0x0,
+			expected: "first_symbol",
+		},
+		{
+			name:     "before_first_symbol",
```

Comment:
This is the exact same `ip` as `exact_match_start` and this test is indeed not before the first symbol